archive/install_shield_3_z.ksy

```diff
@@ -0,0 +1,484 @@
+      - id: reserved
+        contents: [0x00, 0x00, 0x00, 0x00]
+        doc: |
+          Unused according to idecomp.py.
+          Seems to be always 0.
+  version:
+    doc: |
+      File version number,
+      in the same format as used in Windows executable VERSIONINFO resources.
+
+      The version number parts are ordered major.minor.build.private -
+      the strange order of the fields is because Windows normally treats this version format
+      as two little-endian 32-bit integers:
+      the first one with major in the high 16 bits and minor in the low 16 bits,
+      and the second one with build in the high 16 bits and private in the low 16 bits.
+    doc-ref: https://docs.microsoft.com/en-us/windows/win32/menurc/versioninfo-resource
+    -webide-representation: '{major}.{minor}.{build}.{private}'
+    seq:
+      - id: minor
+        type: u2
+      - id: major
+        type: u2
+      - id: private
+        type: u2
+      - id: build
+        type: u2
+  toc_file:
+    seq:
+      - id: end_part
+        type: u1
+        valid:
+          expr: |
+            _root.header.is_extended
+            ? (_ > 0 and _ <= _root.header.num_parts)
+            : _ == 0
+        doc: |
+          Number of the archive part in which the last part of this file's data is stored.
+          Can be the same as `start_part` if this file's data is not split over multiple parts.
+          Only used in extended format -
+          otherwise this field is 0
+          and the file's data is not split.
+      - id: directory_index
+        type: u2
+        doc: |
+          Index into the directory table of contents,
+          indicating in which directory this file is located.
+      - id: len_data_uncompressed
+        type: u4
+        doc: |
+          Byte length of the data after decompression.
+          If the data is not compressed,
+          this is equal to len_data_compressed.
+      - id: len_data_compressed
+        type: u4
+        doc: |
+          Byte length of the data as stored in the archive,
+          i. e. possibly compressed.
+      - id: ofs_data
+        type: u4
+        doc: Absolute byte offset of the file data.
+      - id: modified
+        type: dos_datetime_backwards
+        doc: Modification date/time of the file.
+      - id: attributes
+        type: u4
+        doc: |
+          DOS file attributes of the file.
+
+          Only bits 0 through 5 are actually used by DOS/Windows -
+          all other bits should be zero.
+          Of these,
+          bits 3 and 4 mark volume labels and directories (respectively),
+          so they should never be set on files in an archive.
+
+          For some reason,
+          this field apparently always has bit 7 set if no other attributes are set.
+          This seems to have no meaning and can probably be ignored.
+      - id: len_entry
+        type: u2
+        doc: Total byte size of this file entry.
+      - id: reserved_flags_1
```

Comment:
It may make sense to pack the flags that are packed into a single integer into a separate type to highlight that fact.